src/device-state/preload.ts

```diff
@@ -0,0 +1,114 @@
+import * as _ from 'lodash';
+import { fs } from 'mz';
+
+import { Image } from '../compose/images';
+import DeviceState = require('../device-state');
+
+import constants = require('../lib/constants');
+import { AppsJsonParseError, EISDIR, ENOENT } from '../lib/errors';
+import log from '../lib/supervisor-console';
+
+import { convertLegacyAppsJson } from '../lib/migration';
+import { AppsJsonFormat } from '../types/state';
+
+export async function loadTargetFromFile(
+	appsPath: Nullable<string>,
+	deviceState: DeviceState,
+): Promise<void> {
+	log.info('Attempting to load any preloaded applications');
+	if (!appsPath) {
+		appsPath = constants.appsJsonPath;
+	}
+
+	try {
+		const content = await fs.readFile(appsPath, 'utf8');
+
+		// It's either a tgarget state
```

Comment:
tgarget -> target

Also:

There are two types of people in this world:
1. Those who can extrapolate from incomplete data.